pkg/filter/filter.go

```diff
@@ -48,10 +48,10 @@ func NewFilter(ignoreDBs []string, ignoreTables []TableName, doDBs []string, doT
 func (s *Filter) addOneRegex(originStr string) {
 	if _, ok := s.reMap[originStr]; !ok {
 		var re *regexp.Regexp
-		if originStr[0] != '~' {
-			re = regexp.MustCompile(fmt.Sprintf("(?i)^%s$", originStr))
-		} else {
+		if len(originStr) > 0 && originStr[0] == '~' {
 			re = regexp.MustCompile(fmt.Sprintf("(?i)%s", originStr[1:]))
+		} else { // must match completely
+			re = regexp.MustCompile(fmt.Sprintf("(?i)^%s$", originStr))
```

Comment:
If `originStr` is empty, the regex become `^$`?

Reply:
yes, the else case means you must match totally.
but we check the config for empty schema and table at startup time, so it will not run into this case now actually, because it's meaningless for an empty schema and table, may cause by misconfiguring.